nodes/modifier_make/solidify_mk2.py

```diff
@@ -0,0 +1,169 @@
+                e.verts[0].index+v_len])
+
+    faces_out = [f[::-1] for f in faces] + new_pols + rim_pols
+    pol_group = [0]*len(faces) + [1]*len(new_pols) + [2]*len(rim_pols)
+    new_verts_mask = [0]*len(vertices)+ [1]*len(new_verts)
+
+    edges_out = create_edges(bm, v_len)
+    return (vertices_out, edges_out, faces_out, new_pols, rim_pols, pol_group, new_verts_mask)
+
+class SvSolidifyNodeMk2(bpy.types.Node, SverchCustomTreeNode):
+    """
+    Triggers: Extrude/Thicken Mesh
+    Tooltip: Extrude along normal the mesh surface.
+
+    """
+    bl_idname = 'SvSolidifyNodeMk2'
+    bl_label = 'Solidify'
+    bl_icon = 'MOD_SOLIDIFY'
+
+    thickness: FloatProperty(
+        name='Thickness', description='Shell thickness',
+        default=0.1, update=updateNode)
+    offset: FloatProperty(
+        name='Offset', description='Offset Thickness from center',
+        default=1, soft_min=-1, soft_max=1, update=updateNode)
+    even: BoolProperty(
+        name='Even Thickness', description='Mantain Thinkness by adjusting sharp corners',
+        default=True, update=updateNode)
+
+    def sv_init(self, context):
+        self.inputs.new('SvVerticesSocket', 'Vertices')
+        self.inputs.new('SvStringsSocket', 'Polygons')
+        self.inputs.new('SvStringsSocket', 'Thickness').prop_name = 'thickness'
+        self.inputs.new('SvStringsSocket', 'Offset').prop_name = 'offset'
+
+        self.outputs.new('SvVerticesSocket', 'Vertices')
+        self.outputs.new('SvStringsSocket', 'Edges')
+        self.outputs.new('SvStringsSocket', 'Polygons')
+        self.outputs.new('SvStringsSocket', 'New Pols')
+        self.outputs.new('SvStringsSocket', 'Rim Pols')
+        self.outputs.new('SvStringsSocket', 'Pols Group')
+        self.outputs.new('SvStringsSocket', 'New Verts Mask')
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, 'even')
+    def process(self):
+        if not any((s.is_linked for s in self.outputs)):
+            return
+
+        if not (self.inputs['Vertices'].is_linked and self.inputs['Polygons'].is_linked):
+            return
+
+        verts = self.inputs['Vertices'].sv_get(deepcopy=False)
+        polys = self.inputs['Polygons'].sv_get(deepcopy=False)
+        thickness = self.inputs['Thickness'].sv_get(deepcopy=False)
+        offset = self.inputs['Offset'].sv_get(deepcopy=False)
+
+        verts_out = []
+        edges_out = []
+        polys_out = []
+        new_pols = []
+        rim_pols = []
+        pols_groups = []
+        new_verts_mask = []
+
+        for v, p, t, o in zip_long_repeat(verts, polys, thickness, offset):
+            res = solidify(v, p, t, o, self.even)
+
+            if not res:
+                return
+            verts_out.append(res[0])
+            edges_out.append(res[1])
+            polys_out.append(res[2])
+            new_pols.append(res[3])
+            rim_pols.append(res[4])
+            pols_groups.append(res[5])
+            new_verts_mask.append(res[6])
```

Comment:
Just more short variant. I'm using it in some nodes.
```Python
        res = []

        for v, p, t, o in zip_long_repeat(verts, polys, thickness, offset):
            res.append(solidify(v, p, t, o, self.even))

        verts_out, edges_out, polys_out, new_pols, rim_pols, pols_groups, new_verts_mask = zip(*res)
```

Reply:
Have you done any performance test? Just wondering because append is such a slow operation...

Reply:
I don't know. Getting maximum speed of every single piece of code is not the way of program optimization anyway. It should be done only in bottleneck places all others should be written with priority of readability.